notebooks/02_Fairing/02_04_fairing_kaniko_cloud_builder.ipynb

```diff
@@ -0,0 +1,263 @@
+    "* Below you set some variables specifying a [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/)\n",
+    "* Kubeflow Fairing provides a utility function to guess the name of your AWS account"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from kubernetes import client as k8s_client\n",
+    "from kubernetes.client import rest as k8s_rest\n",
+    "from kubeflow import fairing   \n",
+    "from kubeflow.fairing import utils as fairing_utils\n",
+    "from kubeflow.fairing.builders import append\n",
+    "from kubeflow.fairing.deployers import job\n",
+    "from kubeflow.fairing.preprocessors import base as base_preprocessor\n",
+    "\n",
+    "# Setting up AWS Elastic Container Registry (ECR) for storing output containers\n",
+    "# You can use any docker container registry istead of ECR\n",
+    "AWS_ACCOUNT_ID=fairing.cloud.aws.guess_account_id()\n",
+    "AWS_ACCOUNT_ID = boto3.client('sts').get_caller_identity().get('Account')\n",
+    "DOCKER_REGISTRY = '{}.dkr.ecr.{}.amazonaws.com'.format(AWS_ACCOUNT_ID, AWS_REGION)\n",
+    "\n",
+    "namespace = fairing_utils.get_current_k8s_namespace()\n",
+    "\n",
+    "logging.info(f\"Running in aws region {AWS_REGION}, account {AWS_ACCOUNT_ID}\")\n",
+    "logging.info(f\"Running in namespace {namespace}\")\n",
+    "logging.info(f\"Using docker registry {DOCKER_REGISTRY}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use Kubeflow fairing to build the docker image\n",
+    "\n",
+    "* You will use kubeflow fairing's kaniko builder to build a docker image that includes all your dependencies\n",
+    "  * You use kaniko because you want to be able to run `pip` to install dependencies\n",
+    "  * Kaniko gives you the flexibility to build images from Dockerfiles"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from kubeflow.fairing.builders import cluster\n",
+    "\n",
+    "# output_map is a map of extra files to add to the notebook.\n",
+    "# It is a map from source location to the location inside the context.\n",
+    "output_map =  {\n",
+    "    \"Dockerfile.model\": \"Dockerfile\",\n",
+    "    \"model.py\": \"model.py\"\n",
+    "}\n",
+    "\n",
+    "preprocessor = base_preprocessor.BasePreProcessor(\n",
+    "    command=[\"python\"], # The base class will set this.\n",
+    "    input_files=[],\n",
+    "    path_prefix=\"/app\", # irrelevant since we aren't preprocessing any files\n",
+    "    output_map=output_map)\n",
+    "\n",
+    "preprocessor.preprocess()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a new ECR repository to host model image\n",
+    "!aws ecr create-repository --repository-name fairing-kaniko --region=$AWS_REGION"
```

Comment:
Remember the internal ticket? If there's no repository, we need to create it? Is that still true? Have you verified that? 

What we want is automatically create the repo in the future I think.

Reply:
Got it, you mean the `Create ECR in faring if fairing-job repository doesn't exist` ticket. The truth is that we create `fairing-job` ecr repo in the `02_01_fairing_introduction` [notebook](https://github.com/aws-samples/eks-kubeflow-workshop/blob/3a1dbfff2024436322a7c8c69b14a534c36659a5/notebooks/02_Fairing/02_01_fairing_introduction.ipynb), then for later notebook, we don't have to create repo again. The issue here is if users don't follow the order of notebook, they'll encounter `ECR repository not exist` issue. From my side, I think we could add logic like:

```python
import boto3
ecr_client = boto3.client('ecr')

if not ecr_client.describe_repositories(repo_name='fairing-job'):
    ecr_client.create_repositories(repo_name='fairing-job')
```

WDYT?

Reply:
BTW, the reason we manually create a `fairing-job` is because of that.  Here is a little bit different, because for fairing job launcher, they use fixed `fairing-job`. Kaniko can use different repo_name. Since you use `fairing-kaniko`, it's a new one and let's keep it there.

One that ticket is resolved, we don't need this code snippet any more. fairing will handle this by default, it will create or reuse existing one.